src/Confluent.Kafka/AdminClientBuilder.cs

```diff
@@ -0,0 +1,130 @@
+
+        /// <summary>
+        ///     The configured statistics handler.
+        /// </summary>
+        internal protected Action<AdminClient, string> StatisticsHandler { get; set; }
+
+        /// <summary>
+        ///     Initialize a new <see cref="AdminClientBuilder" /> instance.
+        /// </summary>
+        /// <param name="config">
+        ///     A collection of librdkafka configuration parameters 
+        ///     (refer to https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md)
+        ///     and parameters specific to this client (refer to: 
+        ///     <see cref="Confluent.Kafka.ConfigPropertyNames" />).
+        ///     At a minimum, 'bootstrap.servers' and must be specified.
+        /// </param>
+        public AdminClientBuilder(IEnumerable<KeyValuePair<string, string>> config)
+        {
+            this.Config = config;
+        }
+
+        /// <summary>
+        ///     Set the handler to call on librdkafka statistics events. Statistics are provided as a JSON formatted string as defined here:
+        ///     https://github.com/edenhill/librdkafka/wiki/Statistics
+        /// </summary>
+        /// <remarks>
+        ///     You can enable statistics and set the statistics interval
+        ///     using the statistics.interval.ms configuration parameter
+        ///     (disabled by default).
+        ///
+        ///     Executes on the poll thread (a background thread managed by
+        ///     the admin client).
+        /// </remarks>
+        public AdminClientBuilder SetStatisticsHandler(Action<AdminClient, string> statisticsHandler)
+        {
+            this.StatisticsHandler = statisticsHandler;
+            return this;
+        }
+
+        /// <summary>
+        ///     Set the handler to call on error events e.g. connection failures or all
+        ///     brokers down. Note that the client will try to automatically recover from
+        ///     errors that are not marked as fatal - such errors should be interpreted
+        ///     as informational rather than catastrophic.
+        /// </summary>
+        /// <remarks>
+        ///     Executes on the poll thread (a background thread managed by the admin
+        ///     client).
+        /// </remarks>
+        public AdminClientBuilder SetErrorHandler(Action<AdminClient, Error> errorHandler)
+        {
+            this.ErrorHandler = errorHandler;
+            return this;
+        }
+
+        /// <summary>
+        ///     Set the handler to call when there is information available
+        ///     to be logged. If not specified, a default callback that writes
+        ///     to stderr will be used.
+        /// </summary>
+        /// <remarks>
+        ///     By default not many log messages are generated.
+        ///
+        ///     For more verbose logging, specify one or more debug contexts
+        ///     using the 'debug' configuration property. The 'log_level'
```

Comment:
👍removed in the other builders as well. this is a hangover from before.

something else worth mentioning is that log_level is not actually exposed on the strongly typed `Config` class (there's are rule in the generator that omits all properties with an underscore, and this is the only property for which this may not be appropriate). My view is this is better since it will prevent people from using it when they probably shouldn't be. They can always specify it with the `Set(string, string)` method if they really want to.

Reply:
Yeah, sounds good. And if it ever would become useful (because we turn up the amount of non-debug logging for whatever reason), we can alias it to `log.level`